### ISSUE TITLE
Can't resolve reference #/definitions/inline_response_200_options

### ISSUE THREAD

**Issue by kingLotzi** [RuntimeTools/chai-openapi-response-validator#8]
Hi, sorry to bother you again, but i'm really interested in your library! 

Here is my new error:
Uncaught AssertionError: expected res to satisfy API spec:
{ message: 'The response was not valid.',
  errors: 
   [ { path: 'options[0]',
       errorCode: '$ref.openapi.responseValidation',
       message: 'options[0] can\'t resolve reference #/definitions/inline_response_200_options' }

Here's my swagger.json:

/.../
    "/products/SNC/options" : {
      "get" : {
        "tags" : [ "owner_dx" ],
        "description" : "Get products options of SNC product",
        "operationId" : "get_snc_products_options",
        "parameters" : [ ],
        "responses" : {
          "200" : {
            "description" : "OK",
            "schema" : {
              "type" : "object",
              "properties" : {
                "options" : {
                  "type" : "array",
                  "items" : {
                    "$ref" : "#/definitions/inline_response_200_options"
                  }
                }
              }
            }
          }
}
/.../
"definitions" : {
"inline_response_200_options" : {
      "properties" : {
        "id" : {
          "type" : "number",
          "example" : 1.0
        },
        "name" : {
          "type" : "string",
          "example" : "AVISOS + CONSULTAS A CEL1,25+IVA)PO"
        },
        "short_name" : {
          "type" : "string",
          "example" : "BASE"
        },
        "summary_display" : {
          "type" : "boolean",
          "example" : true
        },
        "active" : {
          "type" : "boolean",
          "example" : true
        },
        "max_messages" : {
          "type" : "number",
          "example" : 0.0
        },
        "product_id" : {
          "type" : "string",
          "example" : "CC"
        },
        "costs" : {
          "$ref" : "#/definitions/inline_response_200_costs"
        },
        "recharge_amounts" : {
          "$ref" : "#/definitions/inline_response_200_recharge_amounts"
        }
      }
    },
}
/.../

Can I be missing some configuration in the assertion?? 
expect(res).to.have.status(200).and.satisfyApiSpec;

Thanks for all!!!

**Comment by rwalle61** [comment]
Hiya, no worries! 

You've written the assertion correctly, and I can't tell from here what exactly the problem is. Would you mind pasting your entire swagger.json and test file here, so I can recreate it?

(and please format it using `, e.g. 
```
...
```
)

**Comment by kingLotzi** [comment]
Hi, thanks again! 
Unfortunately I can't paste the entire swagger for security but I can re-paste the previous one so you can see it better
{
`"paths" : {
    "/products/SNC/options" : {
      "get" : {
        "tags" : [ "owner_dx" ],
        "description" : "Get products options of SNC product",
        "operationId" : "get_snc_products_options",
        "parameters" : [ ],
        "responses" : {
          "200" : {
            "description" : "OK",
            "schema" : {
              "type" : "object",
              "properties" : {
                "options" : {
                  "type" : "array",
                  "items" : {
                    "$ref" : "#/definitions/inline_response_200_options"
                  }
                }
              }
            }
          },`
{
`"definitions" : {
    "inline_response_200_options" : {
      "properties" : {
        "id" : {
          "type" : "number",
          "example" : 1.0
        },
        "name" : {
          "type" : "string",
          "example" : "AVISOS + CONSULTAS A CEL1,25+IVA)PO"
        },
        "short_name" : {
          "type" : "string",
          "example" : "BASE"
        },
        "summary_display" : {
          "type" : "boolean",
          "example" : true
        },
        "active" : {
          "type" : "boolean",
          "example" : true
        },
        "max_messages" : {
          "type" : "number",
          "example" : 0.0
        },
        "product_id" : {
          "type" : "string",
          "example" : "CC"
        },
        "costs" : {
          "$ref" : "#/definitions/inline_response_200_costs"
        },
        "recharge_amounts" : {
          "$ref" : "#/definitions/inline_response_200_recharge_amounts"
        }
      }
    },
}`
}

**Comment by rwalle61** [comment]
Ok - I'll do my best. I'm trying to recreate your scenario.

In the meantime, is your `swagger.json` OpenAPI 2 or 3? I presume 2, as you're using `definitions` rather than `components`. I'm pretty sure we're compatible with OpenAPI 2, but I'll test it.

(For reference, https://blog.readme.io/an-example-filled-guide-to-swagger-3-2/ shows the difference.
![image](https://user-images.githubusercontent.com/18170169/63042459-f03ce900-bec1-11e9-985b-c7b731023edc.png)
)

**Comment by kingLotzi** [comment]
Hi! Thanks for the trying! 
You are right, it's OpenAPI 2, I'm familiar with the difference between 2 and 3 but for now we can't migrate, so I'm struggling with this.

Hope to hear from you soon! Thanks again.

**Comment by rwalle61** [comment]
Ok when I converted my working OpenAPI 3 to OpenAPI 2, I got the same error as you. 

I'm going to double check, but I think the cause of this problem is with our dependency https://github.com/kogosoftwarellc/open-api/tree/master/packages/openapi-response-validator. 

Several PR's have been merged in their sister packages (e.g. openapi-request-validator) after people reported problems with `$ref` schemas (e.g. https://github.com/kogosoftwarellc/open-api/pull/278, https://github.com/kogosoftwarellc/open-api/pull/293, https://github.com/kogosoftwarellc/open-api/pull/309). 

So I will investigate whether their `openapi-response-validator` package is really working for OpenAPI 2. They do claim to "Supports `$ref` in response schemas i.e. `#/definitions/SomeType`" (so OpenAPI 2), and because they seem to have extensive testing I didn't want to duplicate their tests. But I will now!

If I cannot help them fix the bug, I will fix this chai plugin by using a different validator when connected to OpenAPI 2 files

Sorry for the delay!

**Comment by kingLotzi** [comment]
Alright! I'll investigate a little and if I find something useful I'll reach you with the solution.

Hopefully this will be a simple solution, but let's wait.

Thanks a lot for all the help, let's keep in touch with the advances.

Sorry for the bother!

**Comment by rwalle61** [comment]
Fyi I think I've found a fix, just testing now

**Comment by rwalle61** [comment]
Hi @kingLotzi , I've published the bug fix. Would you mind running `npm i` and checking if the problem persists? If so, feel free to reopen this

**Comment by rwalle61** [comment]
The bug was mine. I was too quick to suspect that it was with our dependency! I thought it likely to be them because we don't actually have much code - we just wrap them so that people can use them in chai.

But I saw that they have tests proving that they work in this OpenAPI 2 scenario. Then I noticed that our code was feeding them `components` rather than `definitions`. Tweaking that fixed the bug, so I extended our existing tests to cover both OpenAPI 2 and 3, so that this doesn't happen again.

Thanks for your patience!
